spec/cloud.go

```diff
@@ -314,9 +314,41 @@ func (g gceMeta) toGeneratorResults() *mackerel.Cloud {
 	return &mackerel.Cloud{Provider: "gce", MetaData: g.toGeneratorMeta()}
 }
 
-// SuggestCustomIdentifier for GCE is not implemented yet
+// SuggestCustomIdentifier the identifier of the GCE VM instance
 func (g *GCEGenerator) SuggestCustomIdentifier() (string, error) {
-	return "", nil
+	identifier := ""
+	err := retry.Retry(3, 2*time.Second, func() error {
+		cl := httpCli()
+		// $ curl -q http://metadata.google.internal/computeMetadata/v1/instance/id -H "Metadata-Flavor: Google"
+		// 8618997027285493035 # This is a unique, numerical ID that is generated by Compute Engine.
+		// https://cloud.google.com/compute/docs/storing-retrieving-metadata#default
+		req, err := http.NewRequest("GET", g.metaURL.String()+"/instance/id", nil)
```

Comment:
I guess instance id is already retrieved in `GCEGenerator#Generate`, since `requestGCEMeta` retrieves GCE metadata recursively.
You can reduce http request to GCE metadata endpoint by sharing response between `#Generate` and `#SuggestCustomIdentifier`.

Reply:
As you said, the metadata retrieval logic is duplicated.
I changed the process to cache the value retrieved by `Generate ()`.